content/blog/2020-11-25-november-20-community-gems.md

```diff
@@ -0,0 +1,195 @@
+  gift- new Bitbucket support for CML!
+picture: 2020-11-25/cover.png
+author: elle_obrien
+commentsUrl: https://discuss.dvc.org/t/october-20-community-gems/535
+tags:
+  - Discord
+  - Gems
+  - CML
+  - Cache
+  - Bitbucket
+---
+
+## DVC questions
+
+### [Q: If I make a new Git branch, how can I synchronize it with DVC?](https://discord.com/channels/485586884165107732/485596304961962003/773498570795778058)
+
+Generally, yes. When you make a new Git branch in your project:
+
+```dvc
+$ git checkout -b <my_great_new_branch>
+```
+
+you'll want to next run
+
+```dvc
+$ dvc checkout
+```
+
+to synchronize your `.dvc` files on that branch. But _did you know_ you can
+automate this with a `post-checkout` Git hook? We've got a hook that executes
+`dvc checkout` whenever you run `git checkout`, so you'll always have the
+correct data file versions. Head to our docs to
+[read up on installing Git hooks into your DVC repository](https://dvc.org/doc/command-reference/install#install)
+so you never forget to `dvc checkout`!
+
+### [Q: I have a big, 100 GB directory. I want to know where the contents are located so I can open them with Spark- is there a way to get the location of my files without caching them locally?](https://discord.com/channels/485586884165107732/485596304961962003/771386223403073587)
+
+For this, we'd recommend the DVC Python API's `get_url` function. For example,
+in a Python script you'd write:
+
+```python
+import dvc.api
+
+resource_url = dvc.api.get_url(
+  "<top-level-directory>",
+  repo="https://github.com/<your-repo>")
+)
+```
+
+This code means the API will return the URL for a file that ends in `.dir`. The
+`.dir` file contains a JSON-formatted table of the hashes and relative paths for
+all the files inside `<top-level-directory>`. You could then parse that file to
+tget the locations of the full directory contents.
+
+Why use the API? It's a good way to access DVC-tracked files without downloading
+them to your workspace! No downloads means nothing in your cache. Read more
+about our
+[Python API and the `get_url` function](https://dvc.org/doc/api-reference/get_url#dvcapiget_url)
+in our docs.
+
+### [Q: Can I have more than one `dvc.yaml` file in my project?](https://discord.com/channels/485586884165107732/563406153334128681/777946398250893333)
+
+_Sort of._ Here's what we mean.
+
+By default, DVC pipelines records all your stages (and their inputs and outputs)
+in a single file, `dvc.yaml`. Per directory, you can have one `dvc.yaml` file.
+If you want to run pipelines in a different folder than your project root, you
+could create another `dvc.yaml` in a subdirectory.
+
+However, `dvc.yaml` is intended to be the only file you need to record and
+reproduce pipelines per directory. Pipelines are designed to have all stages
+stored in the same place, and there's currently no method to rename `dvc.yaml`.
+
+If you really, really, _really_ want to put different stages in different files,
```

Comment:
let's focus on multiple `dvc.yaml`s only. Renaming with `--single-stage` is a hack and is deprecated. Also, author was looking for a solution to split, not rename.

Reply:
Done, I'll remove the `--single-stage` method